WalletWasabi.Fluent/ViewModels/Wallets/Buy/Messages/PayNowAssistantMessageViewModel.cs

```diff
@@ -0,0 +1,130 @@
+using System.Reactive.Linq;
+using NBitcoin;
+using ReactiveUI;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using WalletWasabi.Blockchain.Analysis.Clustering;
+using WalletWasabi.BuyAnything;
+using WalletWasabi.Fluent.Extensions;
+using WalletWasabi.Fluent.Helpers;
+using WalletWasabi.Fluent.Models;
+using WalletWasabi.Fluent.Models.UI;
+using WalletWasabi.Fluent.Models.Wallets;
+using WalletWasabi.Fluent.ViewModels.Navigation;
+using WalletWasabi.Fluent.ViewModels.Wallets.Buy.Workflows;
+using WalletWasabi.Fluent.ViewModels.Wallets.Send;
+using WalletWasabi.Logging;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Fluent.ViewModels.Wallets.Buy.Messages;
+
+#pragma warning disable WW001 // Do not use UiContext or Navigation APIs in ViewModel Constructor
+#pragma warning disable WW002 // Make ViewModel Constructor private
+
+public partial class PayNowAssistantMessageViewModel : AssistantMessageViewModel
+{
+	private readonly Workflow _workflow;
+
+	[AutoNotify] private bool _isBusy;
+
+	public PayNowAssistantMessageViewModel(Workflow workflow, ChatMessage message) : base(message)
+	{
+		if (message.Data is not Invoice invoice)
+		{
+			throw new InvalidOperationException($"Invalid Data Type: {message.Data?.GetType().Name}");
+		}
+
+		_workflow = workflow;
+
+		Invoice = invoice;
+		Amount = invoice.Amount;
+		Address = invoice.BitcoinAddress;
+		IsPaid = invoice.IsPaid;
+		PaymentUrl = invoice.Bip21Link;
+		PayButtonText = IsPaid ? "Paid" : "Pay";
+		UiMessage = message.Text;
+
+		UiContext = UiContext.Default;
+		PayNowCommand = ReactiveCommand.CreateFromTask(PayNowAsync, this.WhenAnyValue(x => x.IsPaid).Select(x => !x));
+	}
+
+	public string PayButtonText { get; set; }
+
+	public string PaymentUrl { get; set; }
+
+	public Invoice Invoice { get; }
+
+	public UiContext UiContext { get; set; }
+
+	public decimal Amount { get; }
+
+	public string Address { get; }
+
+	public bool IsPaid { get; }
+
+	public ICommand PayNowCommand { get; }
+
+	private async Task PayNowAsync()
+	{
+		// TODO @SuperJMN: This is a dirty hack to obtain the current wallet. Please remove it before merging.
+		var walletVm = MainViewModel.Instance.NavBar.SelectedWallet;
```

Comment:
You have UiContext here. That can give you the selected wallet

Reply:
That's what I thought, but UiContext.WalletRepository doesn't provide this info :( @ichthus1604 do you know a better way to do it?

Reply:
@SuperJMN I've pushed a fix for this. Let me know your thoughts ;)

Reply:
> @SuperJMN I've pushed a fix for this. Let me know your thoughts ;)

Wonderfull refactor. OMG :) Thanks!